components/locid_transform/src/fallback/mod.rs

```diff
@@ -215,7 +215,10 @@ impl<'a> LocaleFallbackerBorrowed<'a> {
 }
 
 impl LocaleFallbackerBorrowed<'static> {
-    /// Cheaply converts a `LocaleFallbackerBorrowed<'static>` into a `LocaleFallbacker`.
+    /// Cheaply converts a [`LocaleFallbackerBorrowed<'static>`] into a [`LocaleFallbacker`].
+    /// 
+    /// Note that while the conversion is cheap, [`LocaleFallbacker`] is more expensive to use
```

Comment:
what does more expensive mean here? This is just the DataPayload branch, yes? That's not a huge deal.

Reply:
Oh, I see your other comment. Please include something to that effect.

Reply:
Suggestions?

> Note that while the conversion is cheap, using Foo might inhibit some compile-time optimizations that are possible with FooBorrowed.

Reply:
> Note that while the conversion is cheap, Foo contains branching logic which might inhibit compiler optimizations that are possible with FooBorrowed.

Reply:
For one, the borrowed code paths can be made no-alloc, while `DataPayload` will always pull in alloc.

Reply:
If nothing else, a nice thing about Borrowed is that it has an empty Drop impl, so it's truly as close to zero-cost as we can get for static data like properties. However, that's not relevant to this function, since presumably you're calling this function only if you need the other type.

For the language, how about

> Note: Due to additional indirection, using Foo might inhibit some compile-time optimizations that are possible with FooBorrowed.

Reply:
I think you never *need* the other type. We're only using it in places where we've been too lazy to implement a proper compiled data code path.